File: sso-auth/src/main/java/com/study/infosecurity/SSOByRoles/service/Impl/UserServiceImpl.java
package com.study.infosecurity.SSOByRoles.service.Impl;

import com.study.infosecurity.SSOByRoles.model.dto.User;
import com.study.infosecurity.SSOByRoles.repository.UserRepository;
import com.study.infosecurity.SSOByRoles.service.UserService;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

@Service("userService")
public class UserServiceImpl implements UserService {
    private final UserRepository userRepository;

    @Autowired
    public UserServiceImpl(UserRepository userRepository) {
        this.userRepository = userRepository;
    }

    @Override
    public void save(User user) {
        // TODO: Hash password here
        this.userRepository.save(user);
    }

    @Override
    public User getByUsername(String username) {
        return this.userRepository.findByUsername(username).get(0);
    }
}

File: sso-model/src/main/kotlin/com/study/infosecurity/ssoByRoles/model/service/impl/UserServiceImpl.kt
package com.study.infosecurity.ssoByRoles.model.service.impl

import com.study.infosecurity.ssoByRoles.model.dto.User
import com.study.infosecurity.ssoByRoles.model.repository.UserRepository
import com.study.infosecurity.ssoByRoles.model.security.getHash
import com.study.infosecurity.ssoByRoles.model.service.UserService
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.stereotype.Service

@Service("userService")
class UserServiceImpl @Autowired
constructor(private val userRepository: UserRepository) : UserService {

    override fun save(user: User) {
        user.password = getHash(user.password)
        this.userRepository.save(user)
    }

    override fun getByUsername(username: String): User? {
        val users = this.userRepository.findByUsername(username)
        return if (users.isEmpty()) {
            null
        } else users[0]
    }

    override fun findAll(): List<User> {
        return this.userRepository.findAll()
    }

    override fun getById(id: Long): User? {
        return this.userRepository.getOne(id)
    }
}